lib/course/middleware/unsuccess_middleware.rb

```diff
@@ -0,0 +1,27 @@
+module Course
+  module Middleware
+    class UnsuccessMiddleware
+      attr_reader :app
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        status, headers, body = @app.call(env)
+        if [404, 500].include? status
```

Comment:
ok cool, In the middleware above you extracted '/public' and 'GET' strings, but decided to leave magic numbers here ;)